tools/upgradetool/src/main/java/org/openhab/core/tools/UpgradeTool.java

```diff
@@ -0,0 +1,190 @@
+                    LOGGER.info("{}: already contains a 'unit' metadata, skipping it", itemName);
+                } else {
+                    Metadata metadata = metadataStorage.get("stateDescription:" + itemName);
+                    if (metadata == null) {
+                        LOGGER.info("{}: Nothing to do, no state description found.", itemName);
+                    } else {
+                        String pattern = (String) metadata.getConfiguration().get("pattern");
+                        if (pattern.contains(UnitUtils.UNIT_PLACEHOLDER)) {
+                            LOGGER.info(
+                                    "{}: State description contains unit place-holder '%unit%', check if 'unit' metadata is needed!",
+                                    itemName);
+                        } else {
+                            Unit<?> stateDescriptionUnit = UnitUtils.parseUnit(pattern);
+                            if (stateDescriptionUnit != null) {
+                                String unit = stateDescriptionUnit.toString();
+                                MetadataKey defaultUnitMetadataKey = new MetadataKey("unit", itemName);
+                                Metadata defaultUnitMetadata = new Metadata(defaultUnitMetadataKey, unit, null);
+                                metadataStorage.put(defaultUnitMetadataKey.toString(), defaultUnitMetadata);
+                                LOGGER.info("{}: Wrote 'unit={}' to metadata.", itemName, unit);
+                            }
+                        }
+                    }
+                }
+            }
+        });
+
+        metadataStorage.flush();
+    }
+
+    private static void upgradeJsProfile(String baseDir) {
+        Path linkJsonDatabasePath = Path.of(baseDir, "jsondb", "org.openhab.core.thing.link.ItemChannelLink.json");
+        LOGGER.info("Upgrading JS profile configuration in database '{}'", linkJsonDatabasePath);
+
+        if (!Files.isWritable(linkJsonDatabasePath)) {
+            LOGGER.error("Cannot access link database '{}', check path and access rights.", linkJsonDatabasePath);
+            return;
+        }
+        JsonStorage<ItemChannelLink> linkStorage = new JsonStorage<>(linkJsonDatabasePath.toFile(), null, 5, 0, 0,
+                List.of());
+
+        List.copyOf(linkStorage.getKeys()).forEach(linkUid -> {
+            ItemChannelLink link = Objects.requireNonNull(linkStorage.get(linkUid));
+            Configuration configuration = link.getConfiguration();
+            String profileName = (String) configuration.get(ItemChannelLinkConfigDescriptionProvider.PARAM_PROFILE);
+            if ("transform:JS".equals(profileName)) {
+                String function = (String) configuration.get("function");
+                if (function != null) {
+                    configuration.put("toItemScript", function);
+                    configuration.put("toHandlerScript", function);
```

Comment:
The [original behaviour](https://github.com/openhab/openhab-addons/blob/ab910cca9851ee0c3ac06625ebf1d126073dde6d/bundles/org.openhab.transform.javascript/src/main/java/org/openhab/transform/javascript/internal/profiles/JavaScriptTransformationProfile.java#L88-L90) was to pass through the command back to the handler. I'd suggest either not setting this at all (discard) or assign it to `"|input"` to preserve the same behaviour.